retrieval/discovery/ec2.go

```diff
@@ -59,11 +62,24 @@ func NewEC2Discovery(conf *config.EC2SDConfig) *EC2Discovery {
 	if conf.AccessKey == "" && conf.SecretKey == "" {
 		creds = defaults.DefaultChainCredentials
 	}
+
+	var awsConfig = &aws.Config{
+		Region:      &conf.Region,
+		Credentials: creds,
+	}
+
+	// Adds HTTP proxy to aws config
+	if os.Getenv("HTTP_PROXY") != "" {
```

Comment:
We don't support providing proxies via environment variables, it'd have to come from the config file.